rest_api/tests/api_test/utils.py

```diff
@@ -0,0 +1,387 @@
+            self._sha512 = hashlib.sha512(self.to_cbor()).hexdigest()
+        return self._sha512
+
+
+def create_intkey_transaction(verb, name, value, deps, signer):
+    payload = IntKeyPayload(
+        verb=verb, name=name, value=value)
+
+    # The prefix should eventually be looked up from the
+    # validator's namespace registry.
+    addr = make_intkey_address(name)
+
+    header = TransactionHeader(
+        signer_public_key=signer.get_public_key().as_hex(),
+        family_name='intkey',
+        family_version='1.0',
+        inputs=[addr],
+        outputs=[addr],
+        dependencies=deps,
+        payload_sha512=payload.sha512(),
+        batcher_public_key=signer.get_public_key().as_hex())
+
+    header_bytes = header.SerializeToString()
+
+    signature = signer.sign(header_bytes)
+
+    transaction = Transaction(
+        header=header_bytes,
+        payload=payload.to_cbor(),
+        header_signature=signature)
+
+    return transaction
+
+
+def create_batch(transactions, signer):
+    transaction_signatures = [t.header_signature for t in transactions]
+
+    header = BatchHeader(
+        signer_public_key=signer.get_public_key().as_hex(),
+        transaction_ids=transaction_signatures)
+
+    header_bytes = header.SerializeToString()
+
+    signature = signer.sign(header_bytes)
+
+    batch = Batch(
+        header=header_bytes,
+        transactions=transactions,
+        header_signature=signature)
+
+    return batch
+
+def get_signer():
+    context = create_context('secp256k1')
+    private_key = context.new_random_private_key()
+    crypto_factory = CryptoFactory(context)
+    return crypto_factory.new_signer(private_key)
+
+
+
+def _expand_block(cls, block):
+    """Deserializes a Block's header, and the header of its Batches.
+    """
+    cls._parse_header(BlockHeader, block)
+    if 'batches' in block:
+        block['batches'] = [cls._expand_batch(b) for b in block['batches']]
+    return block
+
+
+def _expand_batch(cls, batch):
+    """Deserializes a Batch's header, and the header of its Transactions.
+    """
+    cls._parse_header(BatchHeader, batch)
+    if 'transactions' in batch:
+        batch['transactions'] = [
+            cls._expand_transaction(t) for t in batch['transactions']]
+    return batch
+
+
+def _expand_transaction(cls, transaction):
+    """Deserializes a Transaction's header.
+    """
+    return cls._parse_header(TransactionHeader, transaction)
+
+
+def _parse_header(cls, header_proto, resource):
+    """Deserializes a resource's base64 encoded Protobuf header.
+    """
+    header = header_proto()
+    try:
+        header_bytes = base64.b64decode(resource['header'])
+        header.ParseFromString(header_bytes)
+    except (KeyError, TypeError, ValueError, DecodeError):
+        header = resource.get('header', None)
+        LOGGER.error(
+            'The validator sent a resource with %s %s',
+            'a missing header' if header is None else 'an invalid header:',

```

Comment:
%s %s is not mapping to any variable here, is this code tested?

Reply:
this utility method is not used yet...just collected for the future